### ISSUE TITLE
Clarification for p:when/p:with-input pipe="" needed

### ISSUE THREAD

**Issue by xml-project** [xproc/3.0-specification#655]
Consider the following fragment:
````
<p:choose>
  <p:with-input><doc /></p:with-input>

  <p:when name="when" test="some-expression">
    <p:with-input pipe="" />
````
We say:
1. "If no context is specified on the p:when, the context of the p:choose is used."
2. On `p:pipe`: "If the step attribute is not specified, it defaults to the step which provides the default readable port. If the port attribute is not specified, it defaults to the primary output port of the step identified (explicitly or implicitly)."

To my reading in the above fragment rule 2 applies, so the default XPath content for step `when` is the default readable port of the p:choose, not the default XPath content defined with `p:choose/p:with-input`. 

Right?

Should we state this explicitly in the prose on p:when?

**Comment by gimsieke** [comment]
I agree with all that you say. The with-input on choose is only for setting the XPath context, not for changing the DRP of the contained when/otherwise clauses. And we should mention this again in a note or so.

**Comment by ndw** [comment]
I agree with your analysis, I don't feel strongly about whether or not it needs to be called out specifically. I wouldn't object to doing so.
